baselines/splunk_risky_command_abuse_ml_detection_baseline.yml

```diff
@@ -0,0 +1,71 @@
+name: Splunk Risky Command Abuse ML Detection Baseline
+id: 273df2f7-643a-451a-8d4d-637e39eadc87
+version: 1
+date: '2022-05-27'
+author: Abhinav Mishra, Kumar Sharad and Xiao Lin, Splunk
+type: Baseline
+datamodel:
+- Splunk_Audit
+description: 'This search is to build a per user risky command abuse detection model using MLTK DensityFunction 
+  algorithm based on audit log data of Splunk apps. The model is built from past 7 days users history of running 
+  search commands containing risky keywords, and aggregate the total search run time for each hour as indicator of 
+  user behavior. The default threshold of outlier boundary is defined as 0.1%. This search should be scheduled to 
+  run at least as frequently as every 7 days to build a recent baseline for each user. The name of ML model 
+  generated is "risky_command_abuse" and should be configured as global (not private) in MLTK app, unless the same 
+  account of training this model will be used to perfeorm inference using this model for anomaly detection. 
+  The corresponding detection of using this model is "Splunk Risky Command Abuse ML Detection". '
+search: '| tstats values(Search_Activity.user) as user, sum(Search_Activity.total_run_time) as run_time count 
+  FROM datamodel=Splunk_Audit.Search_Activity WHERE (Search_Activity.user!="") 
+  AND (Search_Activity.total_run_time>1) AND (earliest=-7d@d latest=now) 
+  AND (Search_Activity.search IN ("*| runshellscript *", "*| collect *","*| delete *", "*| fit *", "*| outputcsv *", 
+  "*| outputlookup *", "*| run *", "*| script *", "*| sendalert *", "*| sendemail *", "*| tscollect *")) 
+  AND (Search_Activity.search_type=adhoc) AND (Search_Activity.user!=splunk-system-user) 
+  BY _time, Search_Activity.user span=1h 
+  | fit DensityFunction "run_time" dist=auto threshold=0.001 show_density=true by "user" into "risky_command_abuse" '
+how_to_implement: This detection depends on MLTK and assumes Splunk audit data model is available. 
+  For large enterprises, training the model might take significant computing resources, and might 
+  need to incresase MLTK DensityFunction settings default values, such as max_fit_time, max_groups, etc.
```

Comment:
I would add a note here that this baseline is meant to work with a detection and reference the detection name. Also I would include a reference to this page for algorithm performance tuning https://docs.splunk.com/Documentation/MLApp/5.3.1/User/Configurefitandapply

Reply:
done